http_server/handlers.go

```diff
@@ -0,0 +1,184 @@
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+}
+
+func (data *serviceData) signUpHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		data.singUpGetHandlers(w, r)
+	case "POST":
+		data.signUpPostHandler(w, r)
+	default:
+		http.Error(w, "405 Method Not Allowed", http.StatusMethodNotAllowed)
+	}
+}
+
+func (data *serviceData) singUpGetHandlers(w http.ResponseWriter, r *http.Request) {
+	t := template.Must(template.ParseFiles("forms.html"))
+	err := t.Execute(w,struct{AccountExists bool}{false})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)	
+	}
+}
+
+func (data *serviceData) signUpPostHandler(w http.ResponseWriter, r *http.Request) {	
+	data.mtx.Lock()
+	defer data.mtx.Unlock()
+
+	if err := r.ParseForm(); err != nil {
+		http.Error(w, "Failed to parse the form: %v. ", http.StatusInternalServerError)
+		return
+	}
+
+	u := userData{
+        username: r.FormValue("username"),
+        email: r.FormValue("email"),
+        password: r.FormValue("password"),
+	}
+
+	confirmPassword := r.FormValue("confirmPassword")
+
+	if u.password != confirmPassword {
+		http.Error(w, "Submitted passwords differ.", http.StatusBadRequest)
+		return
+	}
+
+	if _, taken := data.userMap[u.username]; taken {
+		http.Error(w, "Username already taken, please try again", http.StatusBadRequest)
+		return
+	}
+
+	data.userMap[u.username] = u
+
+	fmt.Fprintf(w, "Successfully signed up, %s! Please sign in", data.userMap[u.username].username)
+}
+
+func (data *serviceData) signInHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		data.signInGetHandler(w, r)
+	case "POST":
+		data.signInPostHandler(w, r)
+	default:
+		http.Error(w, "405 Method Not Allowed", http.StatusMethodNotAllowed)
+	}
+
+}
+
+func (data *serviceData) signInGetHandler(w http.ResponseWriter, r *http.Request) {
+	
+	t := template.Must(template.ParseFiles("forms.html"))
+	err := t.Execute(w,struct{AccountExists bool}{true})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)	
+	}
+}
+
+func (data *serviceData) signInPostHandler(w http.ResponseWriter, r *http.Request) {
+	data.mtx.Lock()
+	defer data.mtx.Unlock()
+
+	if err := r.ParseForm(); err != nil {
+		http.Error(w, "Failed to parse the form: %v. ", http.StatusInternalServerError)
+		return
+	}
+
+	username := r.FormValue("username")
+
+	if _, exists := data.userMap[username]; !exists {
+		http.Redirect(w, r, "%2e%2e%2fsignup", http.StatusSeeOther)
+		return
+	}
+
+	userData := data.userMap[username]
+
+	if userData.password != r.FormValue("password") {
+		return
+	}
+
+	token, _ := rand.Int(rand.Reader, big.NewInt(1000))
+	keyToken := token.String()
+	
+	if _, exists := data.sessionMap[keyToken]; exists {
+		panic("Cryptography failed us!")
+	}
+
+	data.sessionMap[keyToken] = username
+	log.Printf("token: %s, username: %s", keyToken, data.sessionMap[keyToken])
```

Comment:
I'm not sure about the security of this. Session tokens likely should not be logged in plain text.